proposals/0360-opaque-result-types-with-availability.md

```diff
@@ -87,44 +87,139 @@ This is a dead-end for the library author although SE-0244 states that it should
 
 ## Proposed solution
 
-To bridge this usability gap I propose to relax same-type restriction for `return`s inside of availability conditions as follows:
-
-A function returning an opaque result type is allowed to return values of different concrete types from conditionally available `if #available`  branches without any other dynamic conditions, if and only if, all universally available `return` statements in its body return a value of the same concrete type `T`. All returned values regardless of their location must meet all of the constraints stated on the opaque type.
-
-## Detailed design
+To bridge this usability gap, I propose to relax the same-type restriction for functions with `if #available` conditions: if an `if #available` condition is always executed, it can return a different type than the type returned by the rest of the function.
 
 Proposed changes allow to:
 
 * Use multiple different `if #available` conditions to return types based on their availability e.g. from the most to least available.
-* Safely fallback to a **single** universally available type if none of the conditions are met.
+* Safely fallback to a **single** universally available (meaning no availability restrictions) type if none of the conditions are met.
 
-Note that although it is possible to use multiple availability conditions, mixing conditional availability with dynamic checks would result in `return`s being considered universally available. The following declarations would still be unsupported:
+Note that although it is possible to use multiple availability conditions, mixing conditional availability with dynamic checks (`if`, `guard`, `switch`, `for`) would result in `return`s being considered universally available. 
```

Comment:
```suggestion
Because the return type must be decidable without running the code in the function, mixing availability conditions with other conditions (such as `if`, `guard`, or `switch`) removes this special power and requires `return`s in the `if #available` to return the same type as the rest of the function.
```

Reply:
Fixed!